POL-391 Update the AWS policies to support SCP (#636)
flexera/policy_templates: compliance/aws/ecs_unused/README.md

## Input Parameters

> - *Allowed Regions* - A list of allowed regions for an AWS account. Click [here](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using-regions-availability-zones.html#concepts-available-regions) to check regions in AWS and enter the region code. If this field is left empty, then the policy will throw an error.

Reviewer: In chat, @cdwilhelm had mentioned:

> I don't think we should force the user to enter all the regions.  Most users will not need/or want to enter all of the regions

The behavior described here does not seem to have taken that into account. I agree with Curt that a solution where leaving the Allowed Regions list empty means that the policy uses all the regions that it finds using the `DescribeRegions` call. This part of the documentation can then make clear that if SCP is enabled and some regions are disabled there will be failures, so only the actually enabled regions should be entered into the list in that case.

> - *Allowed Regions* - A list of allowed regions for an AWS account. Click [here](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using-regions-availability-zones.html#concepts-available-regions) to check regions in AWS and enter the region code. If this field is left empty, then the policy will throw an error.

Author: @douglaswth, I have already mentioned in the README that if this field is left empty the policy will throw an error.

> - *Allowed Regions* - A list of allowed regions for an AWS account. Click [here](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using-regions-availability-zones.html#concepts-available-regions) to check regions in AWS and enter the region code. If this field is left empty, then the policy will throw an error.

Author: @douglaswth, I thought of adding as mentioned below. Is this fine?.

- *Allowed Regions* - A list of allowed regions for an AWS account. See [Available Regions](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using-regions-availability-zones.html#concepts-available-regions) to check regions in AWS and enter the region code. If SCP is enabled for an AWS account then enter only the enabled regions if in case disabled regions are entered or if this field is left blank then the policy will throw an error.

> - *Allowed Regions* - A list of allowed regions for an AWS account. Click [here](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using-regions-availability-zones.html#concepts-available-regions) to check regions in AWS and enter the region code. If this field is left empty, then the policy will throw an error.

Reviewer: Throwing an error when the list is left empty is exactly the opposite of what Curt and I are suggesting. Forcing users who do not have SCP in place to add every single region into the list in order to get useful functionality is not a good user experience. Only requiring adding regions when SCP is enabled makes a lot more sense.

> - *Allowed Regions* - A list of allowed regions for an AWS account. Click [here](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using-regions-availability-zones.html#concepts-available-regions) to check regions in AWS and enter the region code. If this field is left empty, then the policy will throw an error.

Reviewer 2: @SheikSena  If the user leaves the Allowed Regions empty the policy should run without any error, evaluating all regions.

> - *Allowed Regions* - A list of allowed regions for an AWS account. Click [here](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using-regions-availability-zones.html#concepts-available-regions) to check regions in AWS and enter the region code. If this field is left empty, then the policy will throw an error.

Author: @douglaswth, I made the necessary changes as suggested Can you verify once?. Thanks

> - *Allowed Regions* - A list of allowed regions for an AWS account. Please enter the allowed regions code if SCP is enabled, see [Available Regions](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using-regions-availability-zones.html#concepts-available-regions) in AWS. If in case disabled regions are entered then the policy will throw an error. Leave blank to consider all the regions.

Reviewer: ```suggestion
- *Allowed Regions* - A list of allowed regions for an AWS account. Please enter the allowed regions code if SCP is enabled, see [Available Regions](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using-regions-availability-zones.html#concepts-available-regions) in AWS; otherwise, the policy may fail on regions that are disabled via SCP. Leave blank to consider all the regions.
```

Author: @douglaswth, I did all the required changes

> - *Allowed Regions* - A list of allowed regions for an AWS account. See [Available Regions](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using-regions-availability-zones.html#concepts-available-regions) in AWS and enter the region code. If SCP is enabled for an AWS account, then enter only the enabled regions if in case disabled regions are entered then the policy will throw an error. If this field is left blank, then the policy evaluates all the regions fetched using the **DescribeRegions** API call.

Reviewer: > If SCP is enabled for an AWS account, then enter only the enabled regions if in case disabled regions are entered then the policy will throw an error.

This sentence is really confusing and should probably be rephrased.

Author: @douglaswth, Is this fine

*Allowed Regions* - A list of allowed regions for an AWS account. Please enter the allowed regions code if SCP is enabled, see [Available Regions](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using-regions-availability-zones.html#concepts-available-regions) in AWS. If in case disabled regions are entered then the policy will throw an error. Leave blank to consider all the regions.
